packages/admin/admin/src/error/errordialog/createErrorDialogApolloLink.tsx

```diff
@@ -0,0 +1,72 @@
+import { ServerError } from "@apollo/client";
+import { onError } from "@apollo/client/link/error";
+import { Button, Typography } from "@mui/material";
+import { getReasonPhrase, StatusCodes } from "http-status-codes";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { ErrorType } from "./ErrorDialog";
+import { errorDialogVar } from "./errorDialogVar";
+import { ErrorScope, errorScopeForOperationContext } from "./ErrorScope";
+
+export const createErrorDialogApolloLink = () => {
+    return onError(({ graphQLErrors, networkError, operation }) => {
+        const errorScope = errorScopeForOperationContext(operation.getContext());
+
+        if (errorScope !== ErrorScope.Global) {
+            return;
+        }
+
+        let error: string | string[] | undefined;
+        let errorType: ErrorType | undefined;
+        let title: React.ReactNode | undefined;
+        let userMessage: React.ReactNode | undefined;
+        let httpStatus: string | undefined;
+
+        if (graphQLErrors) {
+            error = graphQLErrors.map(({ message }) => message);
+            errorType = "graphql";
+            title = <FormattedMessage id="comet.errorDialog.graphQLErrors.title" defaultMessage="Server error" />;
+        }
+
+        if (networkError) {
+            // Prioritize GraphQL error over network error
+            if (!error) {
+                error = networkError.message;
+                errorType = "network";
+                title = <FormattedMessage id="comet.errorDialog.networkError.title" defaultMessage="Network error" />;
+            }
+
+            if (networkError.name === "ServerError") {
+                const statusCode = (networkError as ServerError).statusCode;
```

Comment:
I would avoid `as` casts and introduce a typeguard for this kind of check:

```
function isServerError(error: NetworkError | undefined): error is ServerError {
    if (error?.name === "ServerError") {
        return (error as ServerError).statusCode !== undefined;
    }
    return false;
}
```

and then using it here:

```
if (isServerError(networkError)) {
    const statusCode = networkError.statusCode;
    //....
```

Reply:
Done: ee327ad9b6a8020a6c98b19520741f5449a4ba7d